sql/2021/resource-hints/consoleLog_incorrect_type.sql

```diff
@@ -0,0 +1,34 @@
+#standardSQL
+# returns the number of pages which preload a resource of the incorrect script type
+
+# helper to create percent fields
+CREATE TEMP FUNCTION AS_PERCENT (freq FLOAT64, total FLOAT64) RETURNS FLOAT64 AS (
+  ROUND(SAFE_DIVIDE(freq, total), 4)
+);
+
+SELECT
+  client,
+  ARRAY_LENGTH(value) AS freq,
+  COUNT(0) AS total
```

Comment:
Are you looking for the number of pages having this value, or just the number of times it appears in the logs?

If number of pages, you'd need something like `COUNT(DISTINCT url)` where `url` is included in the inner selects below.

I'd suggest renaming this to something like `freq` (frequency) or `pages`, and reserving `total` for the denominator, which you can use to calculate the percentage. So consider something like this:
```suggestion
  COUNT(0) AS freq,
  SUM(COUNT(0)) OVER (PARTITION BY client) AS total,
  COUNT(0) / SUM(COUNT(0)) OVER (PARTITION BY client) AS pct
```